### PR TITLE
feat(wallet): add sign data JSON-RPC method

### LINKED ISSUE #1067: Implement message signing method
The wallet should provide a method to be consume by clients (e.g. Sheikah) in order to support generic message signing. One example use case scenario will be the token claiming process.

### PR DESCRIPTION
**Pull request by mariocao**
This JSON-RPC method `sign_data` is used to sign a generic message by using the wallet's master key.

## Request Params

```
{
  "sessionId": "<session id>",
  "walletId": "<wallet id>",
  "data": "<message to be hashed and signed>",
  "extended_pk": "<true in order to include extended public key>"
}
```

`extended_pk` flag is set to `true` the result will include the extended public key by including the `chaincode` field. This field is considered sensitive because with an extended public key it is possible to derive non-hardened child public keys. 

## Response

```
{
  "chaincode": "<hex string>",
  "public_key": "<hex string in compressed format>",
  "signature": "<hex string>"
}
```

Closes #1067

### CODE REVIEW

File: wallet/src/actors/app/handlers/sign_data.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use serde::{Deserialize, Serialize};
+
+use crate::actors::app;
+use crate::{model, types};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SignDataRequest {
+    session_id: types::SessionId,
+    wallet_id: String,
+    data: String,
```
Comment (tmpolaczyk) [inline wallet/src/actors/app/handlers/sign_data.rs]: Add comment explaining that the data is a string which will be hashed.

Comment (mariocao) [reply wallet/src/actors/app/handlers/sign_data.rs]: Wasn't it clear enough? :rofl: 
I will include it. :+1:

Comment (tmpolaczyk) [reply wallet/src/actors/app/handlers/sign_data.rs]: Ah, sorry, I meant a comment above the field:
```suggestion
    // Message to be signed
    data: String,
```

File: wallet/src/model.rs [hunk 2]
```diff
@@ -48,6 +48,13 @@ pub struct Balance {
     pub total: String,
 }
 
+#[derive(Debug, Serialize)]
+pub struct ExtendedKeyedSignature {
+    pub signature: String,
+    pub public_key: String,
+    pub chaincode: String,
```
Comment (tmpolaczyk) [inline wallet/src/model.rs]: When `extended_pk` is set to false, this field is serialized as `"chaincode":""`. It is possible to not serialize it by using a serde attribute:

```suggestion
    #[serde(skip_serializing_if = "is_default")]
    pub chaincode: String,
```

But you need to define the `is_default` function:

```rust
fn is_default<T: Default + PartialEq>(t: &T) -> bool {
    t == &T::default()
}
```

**Comment by aesedepece** [comment]
Merged as 38838dfb76081df7587028afbb1473146934d523 :heart_decoration: :

**Pull request closed without merging**